src/app/compose/compose.component.ts

```diff
@@ -71,6 +71,8 @@ export class ComposeComponent implements AfterViewInit, OnDestroy, OnInit {
     saveErrorMessage: string;
 
     @Input() shouldExitToTable = false;
+    @Input() shouldExitToDrafts = false;
```

Comment:
It seems like the `@Input()`s are unnecessary here – also for the pre-existing `shouldExitToTable`.

`@Input()` only matters if you want to set it in the template, like `<compose shouldExitToTable="true">`. We don't seem to be doing that (and the quick glance at `git log` indicates that we never did :P): `src/app/compose/draftdesk.component.html` only uses the `model` `@Input`. These are just regular properties of this class, so the `@Input` can go.

I also wonder if maybe it'd simplify things if we just replaced this set of booleans with something like `exitTo: string`? It would cut the amount of properties/methods we need. Food for thought :)

Come to think of it, while Angular's `Router` (surprisingly) doesn't have anything like a "back" method, apparently there's a Location service that we could use here: https://stackoverflow.com/questions/35446955/how-to-go-back-last-page. Perhaps we could just use that (in some cases)? The added benefit would be being able to go back specifically to the contact that summoned us, not just the `/contacts` list. Another bit to think about :)

Reply:
I did wonder if it would be saner to change all "from" places to send a url/route to return to.. I'll look at the Location thingy.